base/sparse/abstractsparse.jl

```diff
@@ -28,3 +28,36 @@ function Base.reinterpret(::Type, A::AbstractSparseArray)
           Try reinterpreting the value itself instead.
           """)
 end
+
+# The following two methods should be overloaded by concrete types to avoid
+# allocating the I = find(...)
+_sparse_findnextnz(v::AbstractSparseArray, i) = (I = find(!iszero, v); n = searchsortedfirst(I, i); n<=length(I) ? I[n] : 0)
+_sparse_findprevnz(v::AbstractSparseArray, i) = (I = find(!iszero, v); n = searchsortedlast(I, i);  n>0          ? I[n] : 0)
+
+function findnext(f::typeof(!iszero), v::AbstractSparseArray, i::Int)
+    j = _sparse_findnextnz(v, i)
+    if j == 0
+        return 0
+    end
+    while !f(v[j])
+        j = _sparse_findnextnz(v, j+1)
+        if j == 0
+            return 0
+        end
+    end
+    return j
+end
```

Comment:
Perhaps the following? :)
```julia

function findnext(f::typeof(!iszero), v::AbstractSparseArray, i::Int)
    j = _sparse_findnextnz(v, i)
    while j != 0 && iszero(v[j])
        j = _sparse_findnextnz(v, j+1)
    end
    return j
end
```

Reply:
Looks good! In the interest of expediency, feel free to commit+merge that if you have time.